src/shape/java/us/ihmc/euclid/shape/convexPolytope/interfaces/ConvexPolytope3DReadOnly.java

```diff
@@ -154,6 +156,52 @@ default void getBoundingBox(BoundingBox3DBasics boundingBoxToPack)
    double getConstructionEpsilon();
 
    /** {@inheritDoc} */
+   
+   
+   
+   @Override
+
+   default int intersectionWith(Point3DReadOnly pointOnLine,
+                                Vector3DReadOnly lineDirection,
+                                Point3DBasics firstIntersectionToPack,
+                                Point3DBasics secondIntersectionToPack)
+   
+   {
+      
+      if (firstIntersectionToPack != null)
+         firstIntersectionToPack.setToNaN();
+      if (secondIntersectionToPack != null)
+         secondIntersectionToPack.setToNaN();
+      
+      int numberOfIntersections = 0;
+      
+      //For each face, we check the possibility of intersection with the line
+      for (int i=0; i< getNumberOfFaces(); i++) {
+         //If there isn't already a intersection found
+         if (firstIntersectionToPack == null) {
+            boolean intersectionWithFace = getFace(i).intersectionBetweenLine3DAndFace3D(pointOnLine,lineDirection, firstIntersectionToPack);
+            if (intersectionWithFace) {
+               numberOfIntersections++;   
+            }
+         }
+         //If there is one intersection found
+         if (firstIntersectionToPack != null && secondIntersectionToPack == null) {
+            boolean intersectionWithFace = getFace(i).intersectionBetweenLine3DAndFace3D(pointOnLine,lineDirection, firstIntersectionToPack);
+            if (intersectionWithFace) {
+               numberOfIntersections++;   
+            }
+         }
+         
+         if (firstIntersectionToPack != null && secondIntersectionToPack != null)
+            break;
+      }
+      
+      return numberOfIntersections;
```

Comment:
Before returning, we need to check the ordering of the 2 intersections to be consistent with the other intersection algorithms